src/main/java/io/jsonwebtoken/JwtBuilder.java

```diff
@@ -345,11 +347,36 @@
      * <p>This is a convenience method: the string argument is first BASE64-decoded to a byte array and this resulting
      * byte array is used to invoke {@link #signWith(SignatureAlgorithm, byte[])}.</p>
      *
+     * <h4>Deprecation Notice: Deprecated as of 0.10.0, will be removed in 1.0.0</h4>
```

Comment:
I notice the deprecation comments say 0.10.0, but this PR seems to be targeting 0.9.1-SNAPSHOT.

If this isn't a mistake, do you mind giving me a brief explanation of your branching/release process (high level enough that it's no more than 2-3 min of your time)?

Reply:
Ah, the PR targets master and master *should* be at `0.10.0-SNAPSHOT` not at `0.9.1-SNAPSHOT` as it currently is.  I need to change that - thanks for the catch.

Technically the deprecation notices themselves can target any patch release (0.9.1, 0.9.2, etc) because they're just JavaDoc changes and the signatures of all _non-internal/private_ API classes/methods are not changing.  That said, it was too much of a hassle to create a separate PR and release just for that so I'm bundling those notifications in the 0.10.0 release.

The other changes associated with this PR have to be 0.10.0 because [semver](https://semver.org/) requires a minor revision increase when new public APIs are introduced (e.g. `Encoder`, `Decoder`, new public methods on `JwtBuilder` and `JwtParser` etc).

So, based on semver, the way I manage all my git repos is:

`master` always reflects the next minor point release baseline (i.e. 'SNAPSHOT' in maven terminology) - 0.10.0-SNAPSHOT, 0.11.0-SNAPSHOT, 1.0.0-SNAPSHOT, etc) and contains all PRs/fixes/whatever pending the Maj.min.0 release.

If a bugfix or patch release is warranted for any `Maj.min.0` release, a branch is created from the `Maj.min.0` release tag with a `MAJ.min.x` naming scheme (suffix is the actual string literal `.x`)

For example, if 0.9.0 has been released, and we need to get a patch out (e.g. upgrading Jackson as I did earlier this week), I create a `0.9.x` branch from the `0.9.0` release tag. and that branch reflects `0.9.1-SNAPSHOT` and contains merged PRs awaiting release.

When 0.9.1 is about to be released, the pending changes also get merged to master (to ensure they're always in all future Maj.min releases).  Then 0.9.1 is released and then the 0.9.x branch reflects `0.9.2-SNAPSHOT` in its pom.xml.

At this point we can either keep the `0.9.x` branch around if we think there might be further patch releases (`0.9.2`, `0.9.3`, etc), or just delete it.  If we delete it and a `0.9.2` release is warranted, we would re-create the `0.9.x` branch based on the `0.9.1` tag.

I tend to prefer to delete `.x` branches after a patch release to keep things clean unless I know there is an immediate need for another follow-up patch very soon.

So in summary:

1. `master` always reflects the staged changes about to go out in the next `Maj.min.0` release and should always reflect `Maj.min.0-SNAPSHOT` in its pom.xml.

2. `Maj.min.x` branches (suffixed with the string literal '`x`') always reflect staged changes about to go out in the next Maj.min.patch release and should always reflect `Maj.min.patch-SNAPSHOT` in its pom.xml.

3. After cutting a `Maj.min.patch` release, those changes should also be merged into master (except for pom.xml changes that violate # 1).

I hope this helps!  Does that make sense?

Reply:
I just converted the previous reply to a wiki entry: https://github.com/jwtk/jjwt/wiki/Release-Code-Management for future reference.